erpnext/controllers/selling_controller.py

```diff
@@ -585,6 +587,37 @@ def validate_items(self):
 		from erpnext.controllers.buying_controller import validate_item_type
 		validate_item_type(self, "is_sales_item", "sales")
 
+	def update_serial_items_table(self, update=False):
+		# To update auto-created serials.
+		if self.serial_items and update:
+			for item in self.items:
+				if item.serial_no:
+					for serial in item.serial_no.split('\n'):
```

Comment:
```suggestion
					for serial in item.serial_no.split('\n'):
```

use utility function for getting a serial no from the string. Every manually handling of this has resulted in bugs in past.